From 7caf70b6cda200e1bad77c26e46e465a4ad76d71 Mon Sep 17 00:00:00 2001
From: Georgy Yakovlev <gyakovlev@gentoo.org>
Date: Mon, 2 Jan 2023 21:57:29 -0800
Subject: [PATCH] dpkg-deb: increase buf lengh in movecontrolfiles

In some cases limit of 200 is too short.
For example, on gentoo we build in /var/tmp/portage (user configurable)

the buf contents end up exactly 201 characters:
e.g.: "mv /long/path /another/long/path && rmdir /yet/another/long/path"

so we only catch it in testsuite and dpkg-deb tests fail sometimes.

Signed-off-by: Georgy Yakovlev <gyakovlev@gentoo.org>
---
 src/deb/extract.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/deb/extract.c b/src/deb/extract.c
index a098539..332c664 100644
--- a/src/deb/extract.c
+++ b/src/deb/extract.c
@@ -53,7 +53,7 @@
 static void
 movecontrolfiles(const char *dir, const char *thing)
 {
-  char buf[200];
+  char buf[512];
   pid_t pid;
 
   sprintf(buf, "mv %s/%s/* %s/ && rmdir %s/%s", dir, thing, dir, dir, thing);
-- 
2.39.0

